### ISSUE TITLE
Script suddenly throwing a "msgRegex: (?:) does not have all mandatory capture groups ('type', 'scope', 'breaking', 'description')"

### ISSUE THREAD

**Issue by fran-ink** [qoomon/git-conventional-commits#164]
Getting a new error with this script today: `msgRegex: (?:) does not have all mandatory capture groups ('type', 'scope', 'breaking', 'description')`

Our config is OOTB, no change at all. So i searched a bit and it very much looks related to #163 

I tried defining my own `msgRegex` with the `validValue` mentioned in the PR (`/^(?<type>\w+)(?:\((?<scope>[^()]+)\))?(?<breaking>!)?:\s*(?<description>.+)/i`) and it still doesn't work.

I think this issue comes from:

```
conventionConfig.msgRegex = RegExp(conventionOverride.commitMessageRegexPattern) || conventionConfig.msgRegex;
```

Even if `conventionOverride.commitMessageRegexPattern` is undefined, the RegExp is never null, so `|| conventionConfig.msgRegex` is unreachable. Or maybe the `|| conventionConfig.msgRegex` should be included in the RegExp?

I could bypass the issue by adding this line in my convention:

`commitMessageRegexPattern: ^(?<type>\w+)(?:\((?<scope>[^()]+)\))?(?<breaking>!)?:\s*(?<description>.+)`

**Comment by klingat** [comment]
+1

**Comment by dvqc** [comment]
Had the same issue just now. Nothing changed in our commit messages format.

**Comment by opauloxavier** [comment]
Same issue here. We have now fixed the version to `2.6.7` to keep the pipelines running.

**Comment by qoomon** [comment]
Should be fixed in version `2.7.1`, sorry again for the inconvenience
